lib/cpp/src/thrift/transport/THeaderTransport.cpp

```diff
@@ -394,15 +411,19 @@ void THeaderTransport::setHeader(const string& key, const string& value) {
   writeHeaders_[key] = value;
 }
 
-size_t THeaderTransport::getMaxWriteHeadersSize() const {
+uint32_t THeaderTransport::getMaxWriteHeadersSize() const {
   size_t maxWriteHeadersSize = 0;
   THeaderTransport::StringToStringMap::const_iterator it;
   for (it = writeHeaders_.begin(); it != writeHeaders_.end(); ++it) {
     // add sizes of key and value to maxWriteHeadersSize
     // 2 varints32 + the strings themselves
     maxWriteHeadersSize += 5 + 5 + (it->first).length() + (it->second).length();
   }
-  return maxWriteHeadersSize;
+  if (maxWriteHeadersSize > std::numeric_limits<uint32_t>().max()) {
```

Comment:
I have built all third party libraries under Visual Studio 2010, 2012, 2013, and 2015 in x86 and x86 mode, in debug and release, in static and dynamic mode, where appropriate.  Then I build thrift in the same configurations.  I will be releasing the windows build instructions and contributed build batch files to the project so that they can be used as a reference on how to build thrift on windows for C++ with all of the features enabled, using cmake.